src/snowflake/snowpark/mock/_functions.py

```diff
@@ -933,6 +933,45 @@ def mock_to_variant(expr: ColumnEmulator):
     return res
 
 
+def _compare(x: Any, y: Any) -> Tuple[Any, Any]:
```

Comment:
Let's use `TypeVar` here to indicate that input and output needs to be the same type.